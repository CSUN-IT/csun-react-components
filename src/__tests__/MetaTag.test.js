import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaTag from '../../src/components/meta/Tag';

configure({ adapter: new Adapter() });

describe('MetaTag', () => {
  it('Should be defined', () => {
    expect(MetaTag).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(<MetaTag text={'Warning'} action={'close'} to={'#'} color={'warning'} />);
    expect(tree).toMatchSnapshot();
  });
});
