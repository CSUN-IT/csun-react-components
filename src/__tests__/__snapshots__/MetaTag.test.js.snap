// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaTag Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaTag
    action="close"
    className=""
    color="warning"
    text="Warning"
    to="#"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Warning",
      "className": " tag tag--warning tag--close",
      "replace": false,
      "to": "#",
    },
    "ref": null,
    "rendered": "Warning",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Warning",
        "className": " tag tag--warning tag--close",
        "replace": false,
        "to": "#",
      },
      "ref": null,
      "rendered": "Warning",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
